src/main/java/io/minepkg/testutils/gui/WClickableLabel.java

```diff
@@ -1,29 +1,33 @@
 package io.minepkg.testutils.gui;
 
 import io.github.cottonmc.cotton.gui.widget.WLabel;
+import io.github.cottonmc.cotton.gui.widget.data.InputResult;
 import net.fabricmc.api.EnvType;
 import net.fabricmc.api.Environment;
 import net.minecraft.client.MinecraftClient;
 import net.minecraft.client.sound.PositionedSoundInstance;
 import net.minecraft.sound.SoundEvents;
+import net.minecraft.text.Text;
 
 public class WClickableLabel extends WLabel {
   private Runnable onClick;
 
-  public WClickableLabel(String s) {
-    super(s);
+  public WClickableLabel(Text text) {
+    super(text);
   }
 
   @Environment(EnvType.CLIENT)
   @Override
-  public void onClick(int x, int y, int button) {
+  public InputResult onClick(int x, int y, int button) {
     super.onClick(x, y, button);
 
     if (isWithinBounds(x, y)) {
       MinecraftClient.getInstance().getSoundManager().play(PositionedSoundInstance.master(SoundEvents.UI_BUTTON_CLICK, 1.0F));
 
-      if (onClick!=null) onClick.run();
+      if (onClick != null) onClick.run();
     }
+
+    return InputResult.PROCESSED;
```

Comment:
`InputResult` signifies whether the input was ignored or processed, though I haven't yet figured out what exactly that implies. (May be important for overlapping GUI elements?)

Reply:
looks good :grin: